src/uploader.js

```diff
@@ -440,6 +437,10 @@ define([
                     data = new FormData();
                 data.append("Filedata", file);
                 uploadController.value.updateMediaPath();
+
+                var newExtension = '.' + file.name.split('.').pop().toLowerCase();
+                uploadController.value.uploadParams.path = uploadController.value.uploadParams.path.replace(/(\.[^/.]+)?$/, newExtension);
```

Comment:
What happens here if the file does not have an extension? My quick test says it will add the full name as the extension:
```js
>> newExtension = "." + "name".split('.').pop().toLowerCase()
".name"
>> "/path/name".replace(/(\.[^/.]+)?$/, newExtension)
"/path/name.name" 
```
Is that expected?

Reply:
It's weird, but I don't think it's harmful. Ultimately the server will reject the file for having an unrecognized extension, which I think is fine, we don't need to accept files without extensions.